edx_django_utils/monitoring/internal/middleware.py

```diff
@@ -462,6 +466,60 @@ def log_corrupt_cookie_headers(self, request, corrupt_cookie_count):
             log.info(piece)
 
 
+class FrontendMonitoringMiddleware:
+    """
+    Middleware for adding the forntend monitoring scripts to the response
+    """
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        response = self.get_response(request)
+
+        if response.status_code != 200 or not response['Content-Type'].startswith('text/html'):
+            return response
+
+        # .. setting_name: OPENEDX_TELEMETRY_FRONTEND_SCRIPTS
+        # .. setting_default: None
+        # .. setting_description: Scripts to inject to response for frontend monitoring, this can
+        #    have multiple scripts as we support multiple telemetry backends at once, so we can
+        #    provide multiple frontend scripts in a multiline string for multiple platforms tracking.
+        #    Best is to have one at a time for better performance.
+        frontend_scripts = getattr(settings, 'OPENEDX_TELEMETRY_FRONTEND_SCRIPTS', None)
+
+        if not frontend_scripts:
+            return response
+
+        if not isinstance(frontend_scripts, str):
+            # Prevent a certain kind of easy mistake.
+            raise Exception("OPENEDX_TELEMETRY_FRONTEND_SCRIPTS must be a string.")
+
+        response.content = self.inject_script(response.content, frontend_scripts)
+        return response
+
+    def inject_script(self, content, script):
+        """
+        Add script to the response, if body tag is present.
+        """
+        body = re.search(_HTML_BODY_REGEX, content, re.IGNORECASE)
+
+        # If there is no body tag in html, don't add monitoring scripts
+        if not body:
+            return content
```

Comment:
In this scenario, can we just insert after the `<head>`, if that's present? If not, we should have a comment explaining why.

Reply:
Updated to insert after `<head>` if `<body>` tag is missing.